core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/instance/ProcessInstanceManagerImpl.java

```diff
@@ -236,116 +226,15 @@ private String getSQLClauseIn(String[] items) {
-      selectQuery += "AND activeState.timeoutStatus = 0 ";
-      selectQuery += "AND instance.instanceId = activeState.instanceId ";
-      selectQuery += "AND activeState.backStatus = 0 ";
-      selectQuery += "AND instance.modelId = ?";
-
-      prepStmt = con.prepareStatement(selectQuery);
-      prepStmt.setString(1, state.getName());
-      prepStmt.setString(2, peasId);
-      rs = prepStmt.executeQuery();
-
-      StringBuilder queryBuf = new StringBuilder();
-      queryBuf.append("SELECT distinct instance ");
-      queryBuf.append("FROM org.silverpeas.core.workflow.engine.instance.ProcessInstanceImpl instance ");
-      queryBuf.append("WHERE instanceId IN LIST(");
-
-      while (rs.next()) {
-        String instanceId = rs.getString(1);
-        queryBuf.append("\"");
-        queryBuf.append(instanceId);
-        queryBuf.append("\"");
-        queryBuf.append(" ,");
-      }
-      queryBuf.append(" \"-1\")");
-
-      query = db.getOQLQuery(queryBuf.toString());
-
-      // Execute the query
-      try {
-        results = query.execute(org.exolab.castor.jdo.Database.READONLY);
-
-        // get the instance if any
-        while (results.hasMore()) {
-          ProcessInstance instance = (ProcessInstance) results.next();
-          instances.add(instance);
-        }
-      } catch (Exception e) {
-        SilverLogger.getLogger(this).error(e.getMessage(), e);
-      }
-
-      db.commit();
-
-      return instances.toArray(new ProcessInstance[instances.size()]);
-    } catch (SQLException se) {
-      throw new WorkflowException("ProcessInstanceManagerImpl.getProcessInstancesInState",
-          "EX_ERR_CASTOR_GET_INSTANCES_IN_STATE", "sql query : " + selectQuery, se);
-    } catch (PersistenceException pe) {
-      throw new WorkflowException("ProcessInstanceManagerImpl.getProcessInstancesInState",
-          "EX_ERR_CASTOR_GET_INSTANCES_IN_STATE", pe);
-    } finally {
-      WorkflowJDOManager.closeDatabase(db);
-      DBUtil.close(rs, prepStmt);
-      DBUtil.close(con);
-    }
-  }
-
-  /**
    * Get the process instances for a given instance id
    * @param instanceId id of searched instance
    * @return the searched process instance
    * @throws WorkflowException
    */
   @Override
   public ProcessInstance getProcessInstance(String instanceId) throws WorkflowException {
-    ProcessInstanceImpl instance;
-
-    Database db = null;
-    try {
-      // Constructs the query
-      db = WorkflowJDOManager.getDatabase();
-      db.begin();
-      instance = (ProcessInstanceImpl) db.load(ProcessInstanceImpl.class, instanceId);
-      db.commit();
-
-      return instance;
-    } catch (PersistenceException pe) {
-      throw new WorkflowException("ProcessInstanceManagerImpl.getProcessInstance",
-          "EX_ERR_CASTOR_GET_INSTANCE", pe);
-    } finally {
-      WorkflowJDOManager.closeDatabase(db);
-    }
+    ProcessInstanceImpl instance = repository.getById(instanceId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "instance". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1488)